src/plugins/intel_gpu/src/graph/impls/ocl/paged_attention.cpp

```diff
@@ -66,10 +67,34 @@ struct paged_attention_impl : multi_stage_primitive<paged_attention> {
         return stage == PagedAttentionStage::MIXED;
     }
 
+    void update_inst_params(primitive_inst& inst) const override {
+        OPENVINO_ASSERT(inst.type() == paged_attention::type_id());
+        OPENVINO_ASSERT(inst.get_impl() == this);
+
+        auto& pa_inst = reinterpret_cast<paged_attention_inst&>(inst);
+        pa_inst.query_block_size = get_query_block_size(PagedAttentionStage::PREFILL);
+        pa_inst.use_micro_sdpa = use_micro_sdpa;
```

Comment:
Can we avoid having implementation specific attributes in primitive_inst? Maybe the code from `paged_attention_inst::on_execute()` can be moved to primitive impl itself?

Reply:
Good idea, thanks! Moved from `_inst` to `_impl`